### ISSUE TITLE
URLs to .html without trailing slash and redirect

### ISSUE THREAD

**Issue by noscript** [chrisvfritz/prerender-spa-plugin#179]
I have some routes that are pretending directly pointing to html files:
```
routes: [ 
...
  'pages/1.html',
  'pages/2.html',
...
]
```
and `prerender-spa-plugin` produces directory structure as:
```
pages/
└─1.html/
  └─index.html
└─2.html/
  └─index.html
```

The problem is that most of static web servers with such structure will issue a redirect from `pages/1.html` to `pages/1.html/`. Is there a way to specify output file per route to get structure like this?
```
pages/
└─1.html
└─2.html
```

**Comment by Tribex** [comment]
Hmm, good idea. Perhaps I could add some logic to avoid outputting routes if there is a file extension on the route?

**Comment by noscript** [comment]
It's hard to say if dot in a route always means a file extension. But for example one could append trailing slash to routes to distinguish directories from files and set `fullPath`:

```
routes: [ 
  '/',
  '/about/',
  '/contact/',
  '/pages/1.html'
],
postProcess (renderedRoute) {
  if (renderedRoute.originalPath.endsWith('/')) {
    renderedRoute.fullPath = enderedRoute.originalPath.slice(0, - 1);
  }
  return renderedRoute;
},
```

**Comment by Tribex** [comment]
@noscript I'd prefer not to do that in order to main backwards compatibility as much as possible.

**Comment by noscript** [comment]
Backwards compatibility with v2.1.0? If so, why is it important?

**Comment by Tribex** [comment]
No, with currently existing 3.x installs.

**Comment by noscript** [comment]
Doesn't your first comment imply changing the existing behavior, which would be worse than breaking backwards compatibility? I understand I will have to wait till v.4.x?

**Comment by Tribex** [comment]
@noscript What I mean is that there are likely far more people using a trailing slash than there are people rendering routes with file extensions. That said, the simplest fix would either be a postProcess, or an additional option to enable the behavior.

**Comment by Tribex** [comment]
@noscript Somehow I completely misread what you posted. Sure, I'll go ahead and add something along the lines of what you were asking for. My apologies. :fearful:

**Comment by noscript** [comment]
Thank you, I'm looking forward to it :)

**Comment by Tribex** [comment]
@noscript Implemented in `3.1.0`, see [Using The postProcess Option](https://github.com/chrisvfritz/prerender-spa-plugin#using-the-postprocess-option).

**Comment by noscript** [comment]
Works perfectly! :+1:
